src/webgl/p5.Camera.js

```diff
@@ -1717,26 +1720,25 @@ p5.Camera.prototype._getLocalAxes = function() {
  * @param {Number} dRadius change in radius
  */
 p5.Camera.prototype._orbit = function(dTheta, dPhi, dRadius) {
+  // Calculate the vector and its magnitude from the center to the viewpoint
   const diffX = this.eyeX - this.centerX;
   const diffY = this.eyeY - this.centerY;
   const diffZ = this.eyeZ - this.centerZ;
-
-  // get spherical coorinates for current camera position about origin
-  let camRadius = Math.sqrt(diffX * diffX + diffY * diffY + diffZ * diffZ);
-  // from https://github.com/mrdoob/three.js/blob/dev/src/math/Spherical.js#L72-L73
-  let camTheta = Math.atan2(diffX, diffZ); // equatorial angle
-  let camPhi = Math.acos(Math.max(-1, Math.min(1, diffY / camRadius))); // polar angle
-
-  let newUpY = this.upY > 0 ? 1 : -1;
-  // add change according to the direction of newupY
-  camTheta += newUpY * dTheta;
-  camPhi += newUpY * dPhi;
-  // if camPhi becomes >= PI or <= 0,
-  // upY of camera need to be flipped to the other side
-  if (camPhi <= 0 || camPhi >= Math.PI) {
-    newUpY *= -1;
-  }
-
+  let camRadius = Math.hypot(diffX, diffY, diffZ);
+  // front vector. unit vector from center to eye.
+  const front = new p5.Vector(diffX, diffY, diffZ).normalize();
+  // up vector. normalized camera's unit vector.
+  const up = new p5.Vector(this.upX, this.upY, this.upZ).normalize(); // y-axis
+  // side vector. Right when viewed from the front
+  const side = new p5.Vector.cross(up, front).normalize(); // x-axis
+  // verticalVector. projection of front vector.
+  const vertical = new p5.Vector.cross(side, up).normalize(); // z-axis
```

Comment:
I think we can save ourselves a few cpu cycles here by not normalizing, if I'm not mistaken? $a \times b = \hat{n} \Vert a \Vert \Vert b \Vert\sin\theta$. Since `up` and `side` are already normalized so $\Vert a \Vert \Vert b \Vert = 1$, and both are perpendicular (since `side` is made via a cross product with `up`) so $\sin\theta = \pm 1$, so the result should be unit length still. There's still floating point error to worry about though, of course, do you notice anything weird happen if we don't normalize this one?

Reply:
I was wrong about that. It certainly didn't need to normalize the "vertical" in that calculation...I'll fix it.